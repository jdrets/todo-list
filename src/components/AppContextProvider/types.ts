import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
