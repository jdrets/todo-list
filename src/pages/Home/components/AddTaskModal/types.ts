export interface AddTaskModalType {
  open: boolean
  onClose: () => void
  showSnackbar: (type: string, message: string) => void
}
