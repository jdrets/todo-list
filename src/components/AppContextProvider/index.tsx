import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
