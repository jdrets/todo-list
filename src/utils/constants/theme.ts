const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
