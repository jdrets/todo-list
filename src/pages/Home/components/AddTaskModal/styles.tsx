import styled from 'styled-components'
import theme from '../../../../utils/constants/theme'

export const ButtonsWrapper = styled.div`
  align-items: center;
  bottom: 0;
  display: flex;
  justify-content: space-between;
  left: 0;
  padding: 24px;
  position: absolute;
  width: 100%;
  
  @media (min-width: ${theme.layout.breakpoints.medium}) {
    margin-top: 24px;
    padding: 0;
    position: inherit;
  }
`

export const FieldsWrapper = styled.div`
  display: grid;
  gap: 12px;
`
