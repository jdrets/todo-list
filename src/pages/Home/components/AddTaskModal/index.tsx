import React, { FunctionComponent, useState } from 'react'

import { AddTaskModalType } from './types'
import { ButtonsWrapper } from './styles'

import TaskService from '../../../../services/taskService'
import SNACKBAR from '../../../../utils/constants/snackbar'
import Button from '../../../../components/Button'
import Modal from '../../../../components/Modal'
import Fields from './components/Fields'

const AddTaskModal: FunctionComponent<AddTaskModalType> = ({
  open,
  onClose,
  showSnackbar
}) => {
  const [fields, setFields] = useState({
    title: null,
    status: null,
    priority: null,
    description: null,
    dueDate: null
  })

  const handleChangeField = ({ target: { value, name } }) => {
    setFields({
      ...fields,
      [name]: value
    })
  }

  const handleConfirm = () => {
    try {
      TaskService.create(fields)
    } catch (error) {
      console.log(error)

      showSnackbar(SNACKBAR.TYPE.ERROR, SNACKBAR.MESSAGE.ERROR)
    }
  }

  const handleCancel = () => {
    setFields(null)
    onClose()
  }

  const isButtonDisabled = () => {
    return !fields.status || !fields.priority || !fields.description || !fields.title
  }

  return (
    <Modal
      title="Add new task"
      open={open}
      onClose={handleCancel}
    >
      <Fields handleChangeField={handleChangeField} />
      <ButtonsWrapper>
        <Button
          variant="white"
          onClick={handleCancel}
        >
          Cancel
        </Button>
        <Button
          onClick={handleConfirm}
          // disabled={isButtonDisabled()}
        >
          Create
        </Button>
      </ButtonsWrapper>
    </Modal>
  )
}

export default AddTaskModal
