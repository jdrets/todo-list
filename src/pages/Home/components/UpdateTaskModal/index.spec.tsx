import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'

import AddTaskModal from '.'
import theme from '../../../../utils/constants/theme'
import AppContextProvider from '../../../../components/AppContextProvider'

const setup = (mockData = {}) => {
  const mock = {
    open: true,
    onClose: jest.fn(),
    showSnackbar: jest.fn()
  }
  return render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <AddTaskModal {...mock} {...mockData} />
      </AppContextProvider>
    </ThemeProvider>
  )
}

describe('<AddTaskModal />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
