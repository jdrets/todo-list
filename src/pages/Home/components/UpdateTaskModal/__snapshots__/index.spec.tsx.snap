// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateTaskModal /> should match with snapshot 1`] = `
<div>
  <div
    class="sc-hKFyIo evrwVN"
  >
    <div
      class="sc-eCApGN yMYBc"
    />
    <div
      class="sc-jSFkmK iQtCIj"
    >
      <div
        class="sc-gKAblj bmjdUa"
      >
        <h2
          class="sc-iCoHVE eWMcWA"
        />
        <button
          class="sc-fujyUd dJpoTo"
          data-testid="modal-close-button"
        >
          <span
            class="sc-dlnjPT feMzXu"
            color="#000000"
            height="22px"
            rotate="0"
            width="22px"
          >
            <svg
              class=""
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="sc-fFSRdu htoAzM"
      >
        <div
          class="sc-pNWxx jdNrxi"
        >
          <div
            class="sc-crzoUp hvGDba"
          >
            Status
             
            <span
              class="sc-dIsAE dxguxs"
            >
              *
            </span>
          </div>
          <div
            class="sc-jrsJCI ujEcj"
          >
            <select
              class="sc-iqAbSa iBUuXo"
              name="status"
              placeholder="Task status"
              required=""
            >
              <option
                value="To do"
              >
                To do
              </option>
              <option
                value="Doing"
              >
                Doing
              </option>
              <option
                value="Done"
              >
                Done
              </option>
            </select>
            <div
              class="sc-kEqYlL kBiCre"
            >
              <span
                class="sc-dlnjPT feMzXu"
                color="#000000"
                height="22px"
                rotate="0"
                width="22px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
                  <path
                    d="M112 184l144 144 144-144"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="48"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-pNWxx jdNrxi"
        >
          <div
            class="sc-crzoUp hvGDba"
          >
            Priority
             
            <span
              class="sc-dIsAE dxguxs"
            >
              *
            </span>
          </div>
          <div
            class="sc-jrsJCI ujEcj"
          >
            <select
              class="sc-iqAbSa iBUuXo"
              name="priority"
              placeholder="Task priority"
              required=""
            >
              <option
                value="Low"
              >
                Low
              </option>
              <option
                value="Medium"
              >
                Medium
              </option>
              <option
                value="High"
              >
                High
              </option>
            </select>
            <div
              class="sc-kEqYlL kBiCre"
            >
              <span
                class="sc-dlnjPT feMzXu"
                color="#000000"
                height="22px"
                rotate="0"
                width="22px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  
                  <path
                    d="M112 184l144 144 144-144"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="48"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-bqGHjH kyHjCt"
        >
          <div
            class="sc-hBMVcZ fXfuSD"
          >
            Due date
             
          </div>
          <input
            class="sc-ksluoS eWuGXS"
            name="dueDate"
            placeholder="Due date"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="sc-bqGHjH kyHjCt"
        >
          <div
            class="sc-hBMVcZ fXfuSD"
          >
            Title
             
            <span
              class="sc-fnVYJo bsxSSw"
            >
              *
            </span>
          </div>
          <input
            class="sc-ksluoS eWuGXS"
            name="title"
            placeholder="Task title"
            required=""
            value=""
          />
        </div>
        <div
          class="sc-bqGHjH kyHjCt"
        >
          <div
            class="sc-hBMVcZ fXfuSD"
          >
            Description
             
            <span
              class="sc-fnVYJo bsxSSw"
            >
              *
            </span>
          </div>
          <textarea
            class="sc-ksluoS VrFCC"
            name="description"
            placeholder="Task description"
            required=""
          />
        </div>
      </div>
      <div
        class="sc-bdnylx fhbphs"
      >
        <div
          class="sc-gtssRu ivXUmL"
        >
          Delete
        </div>
        <div
          class="sc-gtssRu dSSfSs"
        >
          Update
        </div>
      </div>
    </div>
  </div>
</div>
`;
